system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -145,7 +144,20 @@ class TopSitesPerfTimer extends React.Component {
       return;
     }
 
-    this._onNextFrame(this._sendPaintedEvent);
+    // If we've already handled a timestamp, don't do it again
+    if (this._timestampHandled) {
+      return;
+    }
+
+    // And if we haven't, we're doing so now, so remember that.
+    // Even if something goes wrong in the callback, we can't
+    // try again, as we'd be sending back the wrong data, and we
+    // have to do it here, so that other calls to this method
+    // while waiting for the next frame won't also try to handle
+    // handle it.
+    this._timestampHandled = true;
```

Comment:
Yeah, I dunno how it ended up that narrow.  As we all know, 80 columns is the One True Width.  :-D